app/src/main/java/com/android/unio/ui/home/Home.kt

```diff
@@ -1,24 +1,215 @@
+import androidx.navigation.NavHostController
+import com.android.unio.model.event.Event
+import com.android.unio.model.event.EventCard
+import com.android.unio.model.event.EventListViewModel
+import com.android.unio.model.event.EventRepository
+import com.android.unio.model.event.EventRepositoryMock
 import com.android.unio.ui.navigation.BottomNavigationMenu
 import com.android.unio.ui.navigation.LIST_TOP_LEVEL_DESTINATION
 import com.android.unio.ui.navigation.NavigationAction
 import com.android.unio.ui.navigation.Route
+import kotlinx.coroutines.launch
 
-@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Preview(showBackground = true)
 @Composable
-fun HomeScreen(navigationAction: NavigationAction) {
+fun EventListOverviewPreview() {
+
+  val mockEventRepository = EventRepositoryMock()
+
+  val eventListViewModel = EventListViewModel(mockEventRepository as EventRepository)
+
+  HomeScreen(
+      NavigationAction(NavHostController(LocalContext.current)),
+      eventListViewModel = eventListViewModel,
+      onAddEvent = {},
+      onEventClick = {})
+}
+
+@Composable
+fun HomeScreen(
+    navigationAction: NavigationAction,
+    eventListViewModel: EventListViewModel = viewModel(factory = EventListViewModel.Factory),
+    onAddEvent: () -> Unit,
+    onEventClick: (Event) -> Unit
+) {
+  val events by eventListViewModel.events.collectAsState()
+  var selectedTab by remember { mutableStateOf("All") }
+  val density = LocalDensity.current.density
+
+  val coroutineScope = rememberCoroutineScope()
+  val animatablePosition = remember { Animatable(0f) }
+
+  var allTabWidth by remember { mutableStateOf(0.dp) }
+  var followingTabWidth by remember { mutableStateOf(0.dp) }
+  var allTabXCoordinate by remember { mutableStateOf(0f) }
+  var followingTabXCoordinate by remember { mutableStateOf(0f) }
+
+  val horizontalHeaderPadding = 16.dp
+
   Scaffold(
-      content = { Text("Home screen") },
-      modifier = Modifier.testTag("HomeScreen"),
+      floatingActionButton = {
+        FloatingActionButton(onClick = onAddEvent, modifier = Modifier.testTag("event_MapButton")) {
+          Icon(imageVector = Icons.Filled.Add, contentDescription = "Add Event")
+        }
+      },
       bottomBar = {
         BottomNavigationMenu(
             { navigationAction.navigateTo(it.route) }, LIST_TOP_LEVEL_DESTINATION, Route.HOME)
+      },
+      modifier = Modifier.testTag("HomeScreen"),
+      content = { paddingValues ->
+        Column(modifier = Modifier.fillMaxSize().padding(paddingValues).background(Color.Black)) {
+          // Sticky Header
+          Box(
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .background(Color.Black)
+                      .padding(vertical = 16.dp, horizontal = horizontalHeaderPadding)
+                      .testTag("event_Header")) {
+                Column {
+                  Text(
+                      text = "Upcoming Events",
```

Comment:
Hardcoded string. This one should probably be in the XML file "res/values/strings.xml"

Reply:
The Home.kt isn't my work, except for the linking with firebase. It is in green because I  changed the name of the file. But you're right we'll have to correct this